install/resources/upgrade50.php

```diff
@@ -35,3 +35,34 @@ function upgrade50_dbchanges()
 	$output->print_contents("<p>Click next to continue with the upgrade process.</p>");
 	$output->print_footer("50_done");
 }
+
+function upgrade50_verify_adminemail()
+{
+	global $output, $cache, $db, $mybb;
+
+	$output->print_header("Verifying Admin Email");
+	if(empty($mybb->settings['adminemail']))
+	{
+		echo "<p>Updating admin email into settings...</p>";
+		echo "<p><small>P.D: Field can not be empty</small></p>";
+		if($mybb->user['email'] != "")
+		{
+			$db->update_query('settings', array('value' => $mybb->user['email']), "name='adminemail'");
+		}
+		else
+		{
+			$query = $db->simple_select('users','email',"usergroup=4 AND email<>''", array("limit" => 1,"order_by" => "id","order_dir" => "asc"));
+			if($db->num_rows($query) == 1)
+			{
+				$email = $db->fetch_field($query, 'email');
+				$db->update_query('settings', array('value' => $email), "name='adminemail'");
+			}
+		}
```

Comment:
It looks like there's a case where the email may still be empty after this - I'm not sure how we want to handle that..?

Reply:
I have seen multiple boards where admin accounts are with empty emails due maybe db manipulation.

I think that is unnecessary anyways but i add only a case just to add a valid email if current user who runs the update don't have that field and go throught the update without any other issue. I think that part is not necessary at all. And with only run the update must be enought to add current user email :) Im that file as i try to set a comment yeah it needs a deep review from you due the people who knows more about the mybb system is the staff i think. But yeah this was a sugestion only on the end to try to help you in some way because you have given to much to all of us and is what i think we have to do at least, try to contribute.

Reply:
I think @euantorano means that administrators can empty the setting after updating. This shouldn't be allowed.

Reply:
> I think @euantorano means that administrators can empty the setting after updating. This shouldn't be allowed.

No I don't think so.

This query `$query = $db->simple_select('users','email',"usergroup=4 AND email<>''", array("limit" => 1,"order_by" => "id","order_dir" => "asc"));` may return no result, so the `adminemail` will still be empty.